Catappult/appcoins-wallet-android
# [APPC-2325] Subs item already owned

**What does this PR do?**

Adds the sub already owned error to all payments (even the ones who don't support subs)
**Important:** This required extra refactor on how we were handling the service errors and on how we were handling the initial checks made on PaymentMethodsPresenter. Altough after this we should be able to map errors easier in the future (except for APPC method, that still has way to many flows).
With this refactor we should now be able to have more details on errors on rakam, since we are now logging only the text part and not everything like we were before.
Because of this we should also add more detail to the ticket description so that the QA team can give a better test to it.
Also added tests to UserSubscriptions Repo, mapper and interactor

**Database changed?**

No

**Where should the reviewer start?**

PaymentMethodsPresenter.kt
ErrorMapper.kt

**How should this be manually tested?**

Do a sub on Trivial Drive, wait 3 minutes (so that TD doesn't recognize the sub anymore, you can change the time on the project if you want). Try to do a purchase, it should display the error at the start.
Do the same with CC as pre selected and check if it's the same behavior
Remove the checkSubscriptionOwned() from PaymentMethodsPresenter.kt and try to do a purchase again, check if the message displayed after inserting the CC is the correct one.
Give a small overview on some error flow. 

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/APPC-2325

**Code Review Checklist**
